ui/src/tasks/components/TaskRunsPage.tsx

```diff
@@ -1,25 +1,63 @@
 // Libraries
 import React, {PureComponent} from 'react'
-import {withRouter, WithRouterProps} from 'react-router'
+import {connect} from 'react-redux'
 
 // Types
-import {Run} from '@influxdata/influx'
 import {Page} from 'src/pageLayout'
 import TaskRunsList from 'src/tasks/components/TaskRunsList'
+import {AppState} from 'src/types/v2'
 
-// DummyData
-import {taskRuns} from 'src/tasks/dummyData'
+// Actions
+import {getRuns} from 'src/tasks/actions/v2'
+import {Run} from '@influxdata/influx'
+import {RemoteDataState} from 'src/types'
 
-interface Props {
+interface PassedInProps {
   params: {id: string}
+}
+
+interface ConnectedDispatchProps {
+  getRuns: typeof getRuns
+}
+
+interface ConnectedStateProps {
   runs: Run[]
 }
 
-const dummyData = taskRuns
+interface State {
+  loading: RemoteDataState
+}
+
+type Props = PassedInProps & ConnectedDispatchProps & ConnectedStateProps
 
-class TaskRunsPage extends PureComponent<Props & WithRouterProps> {
+class TaskRunsPage extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {loading: RemoteDataState.NotStarted}
+  }
   public render() {
-    const {params} = this.props
+    const {params, runs} = this.props
+
+    if (this.state.loading !== RemoteDataState.Done) {
```

Comment:
Could we display a spinner while this is loading? The lack of feedback seems disorienting.

Reply:
Added the spinner